components/gitpod-cli/cmd/validate.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/spf13/cobra"
+	"github.com/xeipuuv/gojsonschema"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var validateCommand = &cobra.Command{
+	Use:   "validate",
+	Short: "Validate your .gitpod.yml file is valid or not",
+	Run: func(_ *cobra.Command, _ []string) {
+
+		// Load .gitpod.yml schema
+		schemaLoader := gojsonschema.NewReferenceLoader("https://gitpod.io/schemas/gitpod-schema.json")
+
+		// Load .gitpod.yml file
+		filePathOfGitpodConfig := os.Getenv("GITPOD_REPO_ROOT") + "/.gitpod.yml"
```

Comment:
only supervisor can make use of env vars, all other clients should use supervisor, they are implementation details of supervisor